StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/SavedPaymentMethods/SavedPaymentMethodsSheet.swift

```diff
@@ -0,0 +1,233 @@
+                let error = PaymentSheetError.unknown(debugDescription: "Failed to retrieve PaymentMethods for the customer")
+//                let error = error ?? PaymentSheetError.unknown(
+//                    debugDescription: "Failed to retrieve PaymentMethods for the customer"
+//                )
+                completion(.failure(.errorFetchingSavedPaymentMethods(error)))
+                return
+            }
+            let filteredPaymentMethods = paymentMethods.filter { $0.type == .card }
+            completion(.success(filteredPaymentMethods))
+        }
+
+    }
+
+}
+
+@available(iOSApplicationExtension, unavailable)
+@available(macCatalystApplicationExtension, unavailable)
+extension SavedPaymentMethodsSheet: SavedPaymentMethodsViewControllerDelegate {
+    func savedPaymentMethodsViewControllerShouldConfirm(_ savedPaymentMethodsViewController: SavedPaymentMethodsViewController, with paymentOption: PaymentOption, completion: @escaping (SavedPaymentMethodsSheetResult) -> Void) {
+        guard let intent = savedPaymentMethodsViewController.intent else {
+            assert(false, "Payment intent not available")
+            completion(.failed(error: SavedPaymentMethodsSheetError.unknown(debugDescription: "No payment intent available")))
+        }
+        self.confirmIntent(intent: intent, paymentOption: paymentOption) { result in
+            completion(result)
+        }
+    }
+
+    func savedPaymentMethodsViewControllerDidCancel(_ savedPaymentMethodsViewController: SavedPaymentMethodsViewController, completion _completion: @escaping () -> Void) {
+        savedPaymentMethodsViewController.dismiss(animated: true) {
+            _completion()
+            self.completion?()
+        }
+    }
+
+    func savedPaymentMethodsViewControllerDidFinish(_ savedPaymentMethodsViewController: SavedPaymentMethodsViewController, completion _completion: @escaping () -> Void) {
+        savedPaymentMethodsViewController.dismiss(animated: true) {
+            _completion()
+            self.completion?()
+        }
+    }
+}
+
+extension SavedPaymentMethodsSheet: LoadingViewControllerDelegate {
+    func shouldDismiss(_ loadingViewController: LoadingViewController) {
+        loadingViewController.dismiss(animated: true) {
+            self.completion?()
+        }
+    }
+}
+
+@_spi(PrivateBetaSavedPaymentMethodsSheet) extension _stpspmsbeta_STPCustomerContext {
+    /// Returns the selected Payment Option for this customer context.
+    /// You can use this to obtain the selected payment method without loading the SavedPaymentMethodsSheet.
+    public func retrievePaymentOptionSelection(
+        completion: @escaping (SavedPaymentMethodsSheet.PaymentOptionSelection?, Error?) -> Void
+    ) {
+        self.listPaymentMethodsForCustomer { paymentMethods, error in
+            guard let paymentMethods = paymentMethods, error == nil else {
+                // TODO: Pass errors from the customerContext
```

Comment:
Can't remember who wrote this `TODO`, but is this still needed? seems like calling the completion block is sufficient.

Reply:
I think the current implementation is fine, though it seems like we're strangely nesting a PaymentSheet error within the SPMS error:
```
                let error = PaymentSheetError.unknown(debugDescription: "Failed to retrieve PaymentMethods for the customer")
                completion(.failure(.errorFetchingSavedPaymentMethods(error)))
```

Reply:
Hmm Changed this to just:
```
                completion(nil, error)
```
I think this is what we want.